lib/mayaUsd/ufe/UsdUndoAttributesCommands.cpp

```diff
@@ -40,8 +40,9 @@ UsdAddAttributeCommand::Ptr UsdAddAttributeCommand::create(
     const std::string&          name,
     const Ufe::Attribute::Type& type)
 {
-    if (UsdAttributes::canAddAttribute(sceneItem, name, type)) {
-        return std::make_shared<UsdAddAttributeCommand>(sceneItem, name, type);
+    const std::string kUniqueName = UsdAttributes::getUniqueAttrName(sceneItem, name);
+    if (UsdAttributes::canAddAttribute(sceneItem, kUniqueName, type)) {
+        return std::make_shared<UsdAddAttributeCommand>(sceneItem, kUniqueName, type);
```

Comment:
That covers only the command, and not the direct call API. Can you make the name unique in `UsdAttributes::doAddAttribute()` instead?

Reply:
You are right, at first, I tried making the unique name directly in` UsdAttributes::doAddAttribute()`. But I preferred to create the `UsdAddAttributeCommand` with the already unique name, in fact from the constructor the name is saved in the field `_name`. When the command has finished, to get the created attribute, we call `UsdAddAttributeCommand::attribute()` which uses `_name`. In `UsdAttributes::doAddAttribute` I cannot update directly the command `_name`. But I can make the unique name in `UsdAttributes::doAddAttribute` and outside that function add the update for the command field `_name `. I thought that creating the command immediately with the unique `_name` would have been less error-prone for a possible mismatch between what it is created in `doAddAttribute` and what we get from `UsdAddAttributeCommand::attribute()` if we forget in the future to update `_name` after calling `doAddAttribute `.

Reply:
Not very fond of that workflow split between the command and the direct call. Both should provide the same services.

Can you add a `_uniqueName` non-const member to the command that gets set from the uniquely named Attribute returned by `doAddAttribute()`? Then use that in `UsdAddAttributeCommand::attribute()`.

You could also make `_name` non-const since we do not get much value from keeping it intact once the command has been executed.

Reply:
Got it. If it is not a problem I directly updated `_name` which now is non-const. I thought that continuing to have one member for the name would be more clear.